packages/queued-request-controller/src/QueuedRequestController.test.ts

```diff
@@ -83,12 +80,94 @@ describe('QueuedRequestController', () => {
       // Mock requestNext function
       const requestNext = jest.fn(() => Promise.resolve());
 
-      await controller.enqueueRequest(requestNext);
+      await controller.enqueueRequest(buildRequest(), requestNext);
 
       // Expect that the request was called
       expect(requestNext).toHaveBeenCalledTimes(1);
     });
 
+    it('switches network if a request comes in for a different selected chain', async () => {
+      const mockSetActiveNetwork = jest.fn();
+      const { messenger } = buildControllerMessenger({
+        networkControllerGetState: jest.fn().mockReturnValue({
+          ...cloneDeep(defaultNetworkState),
+          selectedNetworkClientId: 'selectedNetworkClientId',
+        }),
+        networkControllerSetActiveNetwork: mockSetActiveNetwork,
+      });
+      const options: QueuedRequestControllerOptions = {
+        messenger: buildQueuedRequestControllerMessenger(messenger),
+      };
+      const controller = new QueuedRequestController(options);
+
+      await controller.enqueueRequest(
+        {
+          ...buildRequest(),
+          networkClientId: 'differentNetworkClientId',
+        },
+        () => new Promise((resolve) => setTimeout(resolve, 10)),
```

Comment:
is the setTimeout required? Would a plain `Promise.resolve()` work instead?

Reply:
I don't think it is required in this case. The 10ms interval was copied+pasted from other tests, it seemed to be the default next function that was used throughout. I was just following the convention used previously, I probably would have used a plain `jest.fn().mockResolvedValue()` otherwise.

I could remove the timeout where it's not needed, if you think that would improve things? The 10ms wait is a more realistic mock at least, so I didn't bother with it.